### DISCUSSION TITLE
Setting up custom hostname for master/worker node for UPI/Bare metal installation

### DISCUSSION THREAD

**Discussion by kalgary** [okd-project/okd#981] [Q&A]
Hi,
Is it possible to use a custom hostname for master/worker node in the UPI/baremetal installation of OKD for self provisioned VMs?
As per organization standard, we cannot cjhange the standard naming convention for the hostname of master/worker node. Is it possible to Hardcode the hostname in the ignition files during the installation?

Any helpy would be appreciated!

**Comment by DavidHaltinner** [comment]
I havent done a baremetal install in some time now, but last time I had the ignition files first source the master/worker ignition:
ignition:
  config:
    merge:
      - source: http://blahblah.com/master-or-worker.ign

then I specified the hostname with:
storage:
  files:
    - path: /etc/hostname
      mode: 0644
      overwrite: true
      contents:
        inline: coolname.domain.com

Things could be different nowadays for the baremetal install since I last did a new cluster

**Comment by rvanderp3** [comment]
You should be able to configure a PTR record which defines the desired node name for each of the node IPs prior to the nodes joining the cluster.  The node names can be any naming convention you desire.   Setting the hostname via ignition may work as well, but it seems like the hostname is reverted at some point after ignition but I could be off on that.

**Comment by kalgary** [comment]
Hi 
thanks for the reply.

I can set the PTR for the nodes : example xdfc-01.coolcloud.com, xdfc-02.coolcloud.com etc ...
Every node has already an assigned hostname and DNS after boot that follow the internal standard.

But, how can nodes join the cluster as masters or workers ?

Should I modify the ignition or the install-config.yaml for the control plane and the compute node?
Or, is it enough to leave it this way and use the generated ignition as they are?
```
compute: 
- hyperthreading: Enabled 
  name: worker
  replicas: 0 
controlPlane: 
  hyperthreading: Enabled 
  name: master
  replicas: 3 
```

**Reply by rvanderp3** [reply]
You can use the generated ignitions as they are.  You'll just use the master.ign for the masters and worker.ign for the workers.  The machine config server(as started by the bootstrap node) will hand out the master ignition at a specific URL(it's embedded in those files) and once the cluster bootstraps, the worker URL will become active.  You'll see 503s from the machine config server until the worker URL becomes ready.

**Comment by kalgary** [comment]
Hi @rvanderp3 ,
since we do a bare metal installation, the ignition is passed to each node type at bootstrap time.

Ok, if I understood it correctly, if nodes get a correct PTR which matches their hostname.domain we do not need extra modifications to the ignitions generated from the  install-config.yaml:

```
compute: 
- hyperthreading: Enabled 
  name: worker
  replicas: 0 
controlPlane: 
  hyperthreading: Enabled 
  name: master
  replicas: 3 

```
We can just spin up VMs and provide them the correspondent ignition.
